src/Megamix/Patches.cpp

```diff
@@ -94,29 +83,109 @@ namespace Megamix::Patches {
         return cond | cmp_imm_base | reg | value;
     }
 
+    void AddExtraRowsToFront() {
+        std::vector<MuseumRow>      extraMuseumRows {};
+        std::vector<MuseumRowColor> extraMuseumRowColors {};
+
+        auto PushExtraRow = [&](std::array<u16, 5> newRowIds) {
+            extraMuseumRows.emplace_back(newRowIds, "", 0, 0);
+            extraMuseumRowColors.emplace_back(0x424242ff, 0x00000000);
+        };
+
+        // museum don't support rows with only 2 games
+        if (config->tickflows.size() == 2) {
+            for (auto &pair : config->tickflows) {
+                PushExtraRow({ pair.first, 0x101, 0x101, 0x101, 0x101 });
+            }
+        } else {
+            std::array<u16, 5> newRowIds { 0x101, 0x101, 0x101, 0x101, 0x101 };
+            size_t newRowLength = 0;
+
+            for (auto &pair : config->tickflows) {
+                newRowIds[newRowLength] = pair.first;
+                newRowLength += 1;
+
+                if (newRowLength == 5) {
+                    PushExtraRow(newRowIds);
+
+                    newRowIds = { 0x101, 0x101, 0x101, 0x101, 0x101 };
+                    newRowLength = 0;
+                }
+            }
+
+            // museum don't support rows with only 2 games
+            if (newRowLength == 2) {
+                auto &last_extra_row = extraMuseumRows.back();
+
+                newRowIds[2] = last_extra_row.gameIndices[4];
+
+                last_extra_row.columnCount -= 1;
+                last_extra_row.gameIndices[4] = 0x101;
+            }
```

Comment:
should set newRowLength to 0 here or make the next if an else if

Reply:
this doesn't do any pushing, it only modifies the row that is gong to be added, and the last row that was just added.

Reply:
is that second part of code *needed* to run then? if it's not better to add the else if anyway just in case it might introduce any bugs

Reply:
yea it does, the `if newRowLength != 0` statment is what actually pushes the row if there is any remainders, the `if newRowLength == 2` takes care of the edge case by moving a game from the last pushed row into the one that's about to get added